Doc 12037 vector search beta2

ElliotFrancisHunter opened the pull request:
This is a PR for Beta 2 of Vector Search - Java Desktop

This also includes some QoL changes, fixes and some shifts in requirements. 

- Remove reference to numbered beta outside of downloads
- Better attribute tags to support vector search versioning
- Updated iOS copy as errors were found in the snippets, also replaced inline code with includes to the now correctly tagged snippets
- What's new for the feature
- Adding admonition info for AVX2 instruction set requirements
- Tidying install instructions for Android and iOS and making the Vector Search instructions more prominent
-  Required copy for Java Desktop module  

Preview link: https://preview.docs-test.couchbase.com/Elvec2/couchbase-lite/3.2/cbl-whatsnew.html

ElliotFrancisHunter committed:
DOC:12037- Update iOS pages to use includes

Changes:
. Changed inline snippets to included snippets
. Changed snippets to updated/fixed snippets from Pasin
. Added tags to code files
. Corrected typo on code filename

TODO:
. Update working with copy to reflect new snippets

ElliotFrancisHunter committed:
DOC-12037: Beta 2 V1

Changes:
. Fixed bug in android docs that redirected to Swift page instead
. Added .Net and Java Desktop pages
. Working with Vector Search - Java page complete
. Added new pages to nav and changed .Net and Java nav structure to the new format
. Updated what's new to reflect changes

TODO:
. Add .Net samples
. Probably add Beta 2 section to what's new
. Update snippets copy to reflect any code changes (iOS)

ElliotFrancisHunter committed:
Merge branch 'release/3.2' into DOC-12037-vector-search-beta2

ElliotFrancisHunter committed:
DOC-12037: Formatting tidy

Remove .Net doc implementation for now, pages and nav. Split into separate branch for a beta-3. Changed dropdown and references from beta-x to just Beta unless it's a download binary version.

ElliotFrancisHunter committed:
DOC-12037: beta-2 doc refresh and AVX2 notice

Add the notice for users working with x86_64 processors.
Added tags to some attributes for better versioning
Removed some hard coded values
Removed legacy workaround from android install
Improved formatting of android snippets

ElliotFrancisHunter committed:
DOC-12037: iOS snippet update

Reword snippet copy so it is more in line with updated and fixed iOS snippets.

ElliotFrancisHunter committed:
DOC-12037: final java desktop additions

Add the content announcement to what's new page.
Add java AVX2 prereqs and add the install instructions
Add attribute definitions for vector search in the java module

ElliotFrancisHunter committed:
DOC-12037: Addressing feedback

Corrected AVX2 wording.

Made vector search extension instructions clearer by separation where possible and making admonition more prominent.

bfavini commented on modules/ROOT/pages/cbl-whatsnew.adoc:
```diff
@@ -17,6 +17,39 @@ NOTE: Couchbase Lite 3.0 introduces some breaking changes. +
+
+[IMPORTANT] 
+--
+Vector Search is only available for 64-bit architectures and
```
Small edit to change the placement of the word "only" (Pet peeve :-) ) Also, instead of "machine," should you say "device?"
Vector Search is available only on devices with 64-bit architectures and Intel processors that support the Advanced Vector Extensions 2 (AVX2) instruction set. To verify whether your device supports the AVX2 instruction set, follow these instructions.

ElliotFrancisHunter replied:
Done!

bfavini commented on modules/ROOT/pages/cbl-whatsnew.adoc:
```diff
@@ -84,9 +121,7 @@ xref:3.1@couchbase-lite:ROOT:cbl-whatsnew.adoc[What's new in previous version 3.
-There is no 3.2.0 Beta 1 release for the following platforms:
-
-* Java
+There is currently no 3.2.0 Beta 2 release for the following platforms:
```
The Vector Search beta release is not available for the following platforms:

ElliotFrancisHunter replied:
Done!

bfavini commented on modules/android/pages/gs-install.adoc:
```diff
@@ -143,135 +141,130 @@ dependencies {
 
-Community Edition::
-+
+== Kotlin - Vector Search Extension Install Instructions
```
Instead of "install" instructions, change to "Installation Instructions"...and maybe change the heading to:

Kotlin Vector Search Extension --Detailed Installation Instructions

bfavini commented on modules/android/pages/gs-install.adoc:
```diff
@@ -143,135 +141,130 @@ dependencies {
-+
+== Kotlin - Vector Search Extension Install Instructions
+
+More detailed instructions on getting up and running with the Couchbase Lite Vector Search Extension for Android (Kotlin).
```
Not a complete sentence...Change to something like:

Follow these detailed instructions to get up and running with the Vector Search Extension for Android (Kotlin).

bfavini commented on modules/android/pages/gs-install.adoc:
```diff
@@ -143,135 +141,130 @@ dependencies {
+
+More detailed instructions on getting up and running with the Couchbase Lite Vector Search Extension for Android (Kotlin).
+
+NOTE: The Kotlin download also includes the Java version of Couchbase Lite for Android, along with the other dependencies.
```
Can you actually describe exactly what the download contains? 

The Kotlin download include both the Java version or....

bfavini commented on modules/android/pages/gs-install.adoc:
```diff
@@ -143,135 +141,130 @@ dependencies {
 --
-. Set the Java Version, include the following in the `android {}` section:
+You must have Couchbase Lite installed before you can use the Vector Search Extension.
+Vector Search is only available for 64-bit architectures.
```
Vector Search is available only for devices with 64-bit architectures.

bfavini commented on modules/android/pages/gs-install.adoc:
```diff
@@ -143,135 +141,130 @@ dependencies {
-. Set the Java Version, include the following in the `android {}` section:
+You must have Couchbase Lite installed before you can use the Vector Search Extension.
+Vector Search is only available for 64-bit architectures.
+The Vector Search extension is an *Enterprise only* feature.
```
Enterprise-only

bfavini commented on modules/android/pages/gs-install.adoc:
```diff
@@ -143,135 +141,130 @@ dependencies {
 
---
-
+.. For running on x86_64 architectures, include the following in the `dependencies{}` section instead:
```
For x86_64 architectures:

bfavini commented on modules/android/pages/gs-install.adoc:
```diff
@@ -143,135 +141,130 @@ dependencies {
 ----
 
-. Add {cbl-te} to the `dependencies{}` section:
+. Include the following in the `dependencies{}` section:
```
For architectures other than x86_64:

bfavini commented on modules/android/pages/gs-install.adoc:
```diff
@@ -348,80 +332,71 @@ You must have Couchbase Lite installed before you can use the Vector Search Exte
+=====
 
-  implementation 'com.couchbase.lite:couchbase-lite-android-vector-search-x86_64-1.0.0-{vs-version-maintenance}'
+== Java - Vector Search Extension Install Instructions
```
Java Vector Search Extension: Detailed Installation Instructions

bfavini commented on modules/android/pages/gs-install.adoc:
```diff
@@ -348,80 +332,71 @@ You must have Couchbase Lite installed before you can use the Vector Search Exte
-}
-----
---
+More detailed instructions on getting up and running with the Couchbase Lite Vector Search Extension for Android (Java).
```
change to a complete sentence (see previous comment)

bfavini commented on modules/java/pages/vector-search.adoc:
```diff
@@ -0,0 +1,137 @@
+* Low Latency Application Support - You can run searches locally against a local dataset using a local embedded model. This eliminates the network variability and results in more consistent execution speed.
+Even in the case where the model is not embedded within the local device but is deployed at the edge location, the round trip time (RTT) associated with queries can be significantly reduced compared to searches made over the Internet.
+
+* Cost Per Query Reduction - When you have 100s of 1000s of connected clients querying against a cloud-based LLM, the load on cloud model and operational costs of running the cloud based model can be considerably high.
```
hundreds and hundreds

bfavini reviewed (approved):
Small edits here and there--should not hold up the beta release, but please address eventually.

ElliotFrancisHunter committed:
DOC-12037: Final feedback before release

Polishing feedback ahead of release.
Clarifying need to install CBL AND add vector search to a CBL application to use it.
Correcting some grammar
Tidying the admonitions

ElliotFrancisHunter committed:
DOC-12037: Quick start and link update

Adding quick start information for Java and Android.

Also updated Java API References link to point towards current rather than previous version

ElliotFrancisHunter committed:
DOC-12037: Fix typo for API Reference link

ElliotFrancisHunter committed:
Merge branch 'release/3.2' into DOC-12037-vector-search-beta2
